src/AstRunner/AstMap/AstTokenReference.php

```diff
@@ -0,0 +1,15 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Qossmic\Deptrac\AstRunner\AstMap;
+
+/**
+ * @psalm-immutable
```

Comment:
I don't think we should have this on interfaces as we can only guarantee immutability for the actual implementation.

Reply:
Well having it here enforces immutability on the implementations.

Reply:
https://psalm.dev/r/9f39c9366a